proposed/2023/NuGetAudit-without-nuget.org-as-packageSource.md

```diff
@@ -0,0 +1,168 @@
+<!-- What is the impact of not doing this? -->
+
+### Wait for other servers to implement the VulnerabilityInfo resource
+
+While the NuGet team may be able to influence other teams and encourage them to work on implementing support for the `VulnerabilityInfo` resource, ultimately the teams that implement those NuGet servers have their own backlogs and priorities.
+Therefore, the NuGet team doesn't have agency to ensure that it gets implemented, and customers who want to use *NuGetAudit* may not have enough influence to get the implementation prioritized in a timely manor.
+
+Additionally, not all customers use a server that supports upstreaming, or have configured their server to use nuget.org as an upstream source.
+An example is developers working in highly regulated industries where packages need to be individually vetted and approved before it can be used.
+The reasons customers may do this do not preclude them from using nuget.org as an audit source, so allowing these customers an easy way to use *NuGetAudit* benefits them.
+
+Therefore, in order to provide customers with the ability to use NuGetAudit most easily, it's more effective to implement a new feature that does not depend on other teams to implement features.
+
+### NuGet Team supplied upsourcing server
+
+The NuGet team could provide a web application that looks like a NuGet package source that never contains any packages (and doesn't support push), and the `VulnerabilityInfo` resource effectively proxies nuget.org's VDB.
+However, getting permission to install a new web app is likely difficult in most corporate environments, so wouldn't be practical.
+
+Besides the high up-front cost of implementation, it would also have ongoing maintenance costs to the NuGet team.
+For example ensuring it's patched following every Patch Tuesday, and probably maintaining docker images to make it easy to use.
+
+Unlike this spec's proposal, this vulnerability-only upsourcing server idea could in theory enable customers in companies that block nuget.org at the firewall to get always up-to-date VDB.
+But in such a restrictive environment, I expect it's even less likely that their security would allow this web app to be installed than other companies that simply have a policy of "don't use nuget.org as a package source".
+
+The NuGet team already has a [NuGet server implementation](https://github.com/NuGet/NuGet.Server) that can be self hosted.
+But the V3 NuGet Server API wasn't implemented.
+While I find the idea of a reference implementation of NuGet's server API interesting (imagine ASP.NET Core middleware where customers implement a backend interface, and getting new NuGet features is as simple as upgrading the middleware package version), we should consider [NuGet.Server](https://github.com/NuGet/NuGet.Server)'s lack of modernization as a signal that it's likely an unreasonable cost.
+
+### Read vulnerability database from local files
```

Comment:
Can I read or override with any OSV / OSVDB .json/.zip from a webserver too if a server doesn't implement vulnerability functionality? i.e. 

https://google.github.io/osv.dev/data/#current-data-sources | https://github.com/github/advisory-database#sources

https://storage.googleapis.com/osv-vulnerabilities/index.html?prefix=NuGet/ (see all.zip)

Reply:
Currently, no. The original NuGetAudit spec proposed a custom file format, and I don't remember anybody commenting asking for other formats to be readable.